src/KubeOps/Operator/Entities/Extensions/EntityToCrdExtensions.cs

```diff
@@ -251,6 +248,11 @@ internal static V1CustomResourceDefinition CreateCrd(this Type entityType)
             {
                 props.XKubernetesIntOrString = true;
             }
+            else if (typeof(IKubernetesObject).IsAssignableFrom(type) &&
+                     type.Assembly == typeof(IKubernetesObject).Assembly)
```

Comment:
Am I right to assume, that with this assembly check, you want to exclude all "non standard" kubernetes objects? Meaning, that your own kubernetes objects wont trigger this?

Reply:
Yes, it will result in the custom CRD type itself from being ignored.

I think I will have to limit this to concrete types as the serializer in the k8s client does not do polymorphic deserialization based on the kind. I will submit another commit to address this.